utils/common/src/test/kotlin/ExtensionsTest.kt

```diff
@@ -375,13 +376,21 @@ class ExtensionsTest : WordSpec({
 
             assertSoftly {
                 reserved.forEach {
-                    val hexString = String.format(Locale.ROOT, "%%%02X", it.code)
-                    it.toString().percentEncode() shouldBe hexString
+                    val decoded = it.toString()
+
+                    val encoded = decoded.percentEncode()
+
+                    encoded shouldBe String.format(Locale.ROOT, "%%%02X", it.code)
+                    URLDecoder.decode(encoded, Charsets.UTF_8) shouldBe decoded
                 }
 
                 unreserved.forEach {
-                    val singleCharString = it.toString()
-                    singleCharString.percentEncode() shouldBe singleCharString
+                    val decoded = it.toString()
+
+                    val encoded = decoded.percentEncode()
+
+                    encoded shouldBe decoded
+                    URLDecoder.decode(encoded, Charsets.UTF_8) shouldBe decoded
```

Comment:
To me it seems that the first assertion (line 392 / 383) has become redundant. 
If we extract a function `fun String.decode() = URLDecoder.decode(this, Charsets.UTF_8)`, then the test can be rewritten as a one liner, such as: `original.encode().decode() shouldBe original`

What do you think?

Reply:
Doing so would omit the check that the encoded string looks like expected, i.e. having hex chars, or not being different to the decoded string. However, I want to explicit check for properties of the encoded string.

Reply:
I see, then let's leave it as-is.